src/liballoc/rc.rs

```diff
@@ -182,6 +183,24 @@ pub struct Rc<T> {
     _noshare: marker::NoSync
 }
 
+/// An immutable reference-counted pointer type.
+///
+/// See the [module level documentation](../index.html) for more details.
+#[unsafe_no_drop_flag]
+#[stable]
+#[cfg(not(stage0))] // NOTE remove cfg after next snapshot
+pub struct Rc<T> {
+    // FIXME #12808: strange names to try to avoid interfering with field accesses of the contained
+    // type via Deref
+    _ptr: NonZero<*mut RcBox<T>>,
+}
+
+#[cfg(not(stage0))] // NOTE remove cfg after next snapshot
+impl<T> !marker::Send for Rc<T> {}
+
+#[cfg(not(stage0))] // NOTE remove cfg after next snapshot
+impl<T> !marker::Sync for Rc<T> {}
```

Comment:
I think it's good practice though to be explicit; it serves as documentation. It'd be good to add some comments to the impls though saying *why* `Rc` is not sendable/sync. something like

```rust
/// `Rc` instances cannot be sent to other threads because they contain a shared reference count which is modified without atomic instructions.
impl<T> !marker::Send for Rc<T> { }

/// `Rc` instances cannot be sent to other threads because they contain a shared reference count which is modified without atomic instructions.
impl<T> !marker::Sync for Rc<T> { }
```